### ISSUE TITLE
endless loop at ranlib on Strawberry

### ISSUE THREAD

**Issue by plicease** [plicease/Alien-FFI#1]
```
/bin/sh ./libtool  --tag=CC   --mode=link gcc  -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions   -o libffi_convenience.la  src/prep_cif.lo src/types.lo src/raw_api.lo src/java_raw_api.l
o src/closures.lo       src/x86/ffi.lo src/x86/win64.lo
libtool: link: ar cru .libs/libffi_convenience.a src/.libs/prep_cif.o src/.libs/types.o src/.libs/raw_api.o src/.libs/java_raw_api.o src/.libs/closures.o src/x86/.libs/ffi.o src/x86/.libs/win64.o
libtool: link: ranlib .libs/libffi_convenience.a
libtool: link: ( cd ".libs" && rm -f "libffi_convenience.la" && cp -pR "../libffi_convenience.la" "libffi_convenience.la" )
/bin/sh ./libtool  --tag=CC   --mode=link gcc  -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions -no-undefined -version-info `grep -v '^#' ../libtool-version`  -no-undefined -bindir "N:/h
ome/ollisg/perl5/strawberry/x64/5.20.0/lib/perl5/auto/share/dist/Alien-FFI/bin"  -o libffi.la -rpath N:/home/ollisg/perl5/strawberry/x64/5.20.0/lib/perl5/auto/share/dist/Alien-FFI/lib/../lib src/prep_cif.lo src/types.lo src/r
aw_api.lo src/java_raw_api.lo src/closures.lo       src/x86/ffi.lo src/x86/win64.lo
libtool: link: gcc -shared  src/.libs/prep_cif.o src/.libs/types.o src/.libs/raw_api.o src/.libs/java_raw_api.o src/.libs/closures.o src/x86/.libs/ffi.o src/x86/.libs/win64.o    -O3 -march=core2   -o .libs/libffi-6.dll -Wl,--
enable-auto-image-base -Xlinker --out-implib -Xlinker .libs/libffi.dll.a
libtool: link: ar cru .libs/libffi.a  src/prep_cif.o src/types.o src/raw_api.o src/java_raw_api.o src/closures.o src/x86/ffi.o src/x86/win64.o
libtool: link: ranlib .libs/libffi.a
```

**Comment by aero** [comment]
I have the same problem on strawberry perl <= 5.18.

**Comment by plicease** [comment]
Ugh.  I cannot figure this out, running the identical configure and make from the command line instead of from Perl works fine.  Workarounds for now include:

- upgrade to Strawberry 5.20.1 which comes with libffi (and a non borked PkgConfig.pm)
- install libffi yourself (not too hard with MSYS) using the gcc that comes with strawberry, and add the libffi.pc file to your PKG_CONFIG_PATH (if the details for this are not clear I can probably help)

I would love to fix this though.

**Comment by plicease** [comment]
I think I have figured this out.  I am able to reproduce this endless loop from the command line if I put the MSYS commands at the end of the PATH.  `Alien::MSYS` does this to ensure that the compiler and other tools that come with Strawberry are preferred over the ones that come with MSYS.  In this case this might not be the right thing to do!  I will have a fix in the appropriate place shortly.

**Comment by plicease** [comment]
I have a dev release of Alien::FFI uploaded as 0.04_01.  @aero You may want to try it out, if you do let me know.

**Comment by aero** [comment]
I works well. :)

**Comment by plicease** [comment]
Cool.  Thanks for letting me know @aero.  I am going to push some of the changes back into `Alien::Base` see 

https://github.com/Perl5-Alien/Alien-Base/pull/103

so that the benefits can be reaped elsewhere.

**Comment by plicease** [comment]
This appears to be fixed in `Alien::FFI` 0.05 now on CPAN
